CIP-????/README.md

```diff
@@ -0,0 +1,98 @@
+
+The current implementations for web3 applications use static strings. This is dangerous because if a bad actor intercepts the signed message then it can be used in a replay attack by the bad actor. That's why it is very important to produce a dynamic payload rather a static string.
+
+Another problem with the current approach is how the wallets show the information contained in the payload. The payload is a enconded byte array and it could contain anything. If Alice want to call an endpoint and Bob has the ability to change the message before Alice gets it. Alice must be notified somehow that she is signing a potentially malicious payload. A simple hex-encoded representation of the payload isn't enough to ensure a safe interaction.
+
+## Specification
+
+This specification involves multiple parties. We split this specification in three parts: server processing, payload structure and wallet specification.
+
+### Requirement Levels 
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+### Payload structure
+
+The payload MUST be encoded as a JSON string. JSON strings are semi-structured data that are human-readable. So that with a straightforward decoding into text, it will be understandable for the readers. This feature also allows the users to debug it in an easy manner, for example, with the browser debugger tools.
+
+The content of the payload will be included in the protected header of the COSESign1 signature, hence the content effects directly the behavior and security of the system. The payload MUST have the following fields: 
+
+1. The field ```url``` MUST contain the full path to the endpoint, where the payload will be processed.
+
+2. Sometimes, the endpoint ```url``` field is not enough to determine its purpose. The user should understand perfectly the objective of the payload which he or she is signing. That's why the payload MUST contain an ```action``` field with a descriptive text containing the purpose of the payload. 
+
+3. The payload MUST include a UNIX timestamp. The ```timestamp``` field is used as a nonce and a mark for the payload expiration. This field is very imporant in case the payload is compromised.
+
+Additional fields MAY be included in the payload. Depending on the process, it could be useful to include some aditional fields in the protected header of the signature. For example, the email information should be included in the protected header in a registration request. In that way, this payload can be used only to register that specific email and it can not be tampered. 
+
+```JSON
+{
+    "url": "http://example.com/signup",
+    "action": "Sign up",
+    "timestamp": 1673261248,
+    ["email": "email@example.com"] <- Optional
```

Comment:
From GDPR (europe wise) point of view even putting email here as optional is a total disaster...

Reply:
Maybe it's a bad example. I don't know much about the GDPR. Could you please tell me more about why including the email in the payload can be a problem for the GDPR?

Reply:
Once the email address is in there, it's personally identifiable data that cannot be removed.  According to my understanding, even if an email address is added optionally it also has to be removable later.

- Technical recommendations (with lots of references): https://www.nature.com/articles/s41598-022-19341-y
- Bureaucratic view from EU itself & recommendations: https://www.europarl.europa.eu/RegData/etudes/STUD/2019/634445/EPRS_STU(2019)634445_EN.pdf

Reply:
Once the email address is in there, it's personally identifiable data that cannot be removed.  According to my understanding, even if an email address is added optionally it also has to be removable later: even if you have to destroy the whole blockchain to efface it.

- Technical recommendations (with lots of references): https://www.nature.com/articles/s41598-022-19341-y
- Bureaucratic view from EU itself & recommendations: https://www.europarl.europa.eu/RegData/etudes/STUD/2019/634445/EPRS_STU(2019)634445_EN.pdf

Reply:
Thanks for the reply. Now I see your point but the payload is signed in the client and sent to the server. It isn't stored in the blockchain. So, the GDPR doesn't apply in this case. Right?

Reply:
> @jmagan: It isn't stored in the blockchain
As far as I'm concerned that takes care of it.  Any other situations where personal data might be exposed off-chain can be dealt with later if they're a problem. 😎

Reply:
Great. Then can I mark as resolve?

Reply:
Yes (I'm marking it) 😎